openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTrainer.java

```diff
@@ -231,28 +229,30 @@ private static void initializeLightGBMTrainDatasetFeatures(final SWIGTrainData s
 
         /// First generate the array that has the chunk sizes for `LGBM_DatasetCreateFromMats`.
         final SWIGTYPE_p_int swigChunkSizesArray = genSWIGFeatureChunkSizesArray(swigTrainData, numFeatures);
-
-        /// Now create the LightGBM Dataset itself from the chunks:
-        logger.debug("Creating LGBM_Dataset from chunked data...");
-        final int returnCodeLGBM = lightgbmlib.LGBM_DatasetCreateFromMats(
-                (int) swigTrainData.swigFeaturesChunkedArray.get_chunks_count(), // numChunks
-                swigTrainData.swigFeaturesChunkedArray.data_as_void(), // input data (void**)
-                lightgbmlibConstants.C_API_DTYPE_FLOAT64,
-                swigChunkSizesArray,
-                numFeatures,
-                1, // rowMajor.
-                trainParams, // parameters.
-                null, // No alighment with other datasets.
-                swigTrainData.swigOutDatasetHandlePtr // Output LGBM Dataset
-        );
-        if (returnCodeLGBM == -1) {
-            logger.error("Could not create LightGBM dataset.");
-            throw new LightGBMException();
+        try {
+            /// Now create the LightGBM Dataset itself from the chunks:
+            logger.debug("Creating LGBM_Dataset from chunked data...");
+            final int returnCodeLGBM = lightgbmlib.LGBM_DatasetCreateFromMats(
+                    (int) swigTrainData.swigFeaturesChunkedArray.get_chunks_count(), // numChunks
+                    swigTrainData.swigFeaturesChunkedArray.data_as_void(), // input data (void**)
+                    lightgbmlibConstants.C_API_DTYPE_FLOAT64,
+                    swigChunkSizesArray,
+                    numFeatures,
+                    1, // rowMajor.
+                    trainParams, // parameters.
+                    null, // No alighment with other datasets.
+                    swigTrainData.swigOutDatasetHandlePtr // Output LGBM Dataset
+            );
+            if (returnCodeLGBM == -1) {
+                logger.error("Could not create LightGBM dataset.");
+                throw new LightGBMException();
+            }
+            // FIXME is this init necessary?
```

Comment:
Yes this is necessary. It's in here we pass from a `void **` to a LGBM `DatasetHandle` (https://lightgbm.readthedocs.io/en/latest/C-API.html#c.DatasetHandle) so we can pass it around with SWIG.